Source/hydro/advection_util_nd.F90

```diff
@@ -692,6 +692,13 @@ subroutine ca_srctoprim(lo, hi, &
        n = upass_map(ipassive)
        iq = qpass_map(ipassive)
 
+       ! we already accounted for velocities above
+       if (n == QU .or. n == QV .or. n == QW) cycle
```

Comment:
I think you should be testing on iq here, not n?

Reply:
yes.  Thankfully the velocity indices are the same.  Wonder if this is wrong elsewhere...

Reply:
I checked the other places and it looked okay elsewhere.